rules/TypeDeclaration/TypeAnalyzer/AlwaysStrictScalarExprAnalyzer.php

```diff
@@ -89,7 +88,7 @@ private function resolveCastType(Cast $cast): ?Type
 
     private function isScalarType(Type $type): bool
     {
-        if ($type->isString()->yes() && ! $type instanceof ConstantStringType) {
+        if ($type->isString()->yes()) {
             return true;
         }
```

Comment:
Would this skip the docblock string type?

Reply:
The type definition is native type, see line 71:

https://github.com/rectorphp/rector-src/blob/aee42009d91dfdca128760fc4b37e09e101db23f/rules/TypeDeclaration/TypeAnalyzer/AlwaysStrictScalarExprAnalyzer.php#L81-L82

that's will skip docblock read as type :)

Reply:
also on line 71, use native type as well:

https://github.com/rectorphp/rector-src/blob/aee42009d91dfdca128760fc4b37e09e101db23f/rules/TypeDeclaration/TypeAnalyzer/AlwaysStrictScalarExprAnalyzer.php#L71-L72

Reply:
I vaguely recall PHPStan resolved this incorrectly, once the method call was in external file. Making docblock as native types.

If it works, all good :)

Reply:
for FuncCall, and native Function Reflection, it require `->getType()` instead of `->getNativeType()`, that's expected as PHPStan seems doens't read it based on defined Func Call, see

https://github.com/rectorphp/rector-src/blob/aee42009d91dfdca128760fc4b37e09e101db23f/rules/TypeDeclaration/TypeAnalyzer/AlwaysStrictScalarExprAnalyzer.php#L139-L140